src/app/sessions/book/page.tsx

```diff
@@ -32,8 +80,21 @@ export default function BookSessionPage() {
       if (response.ok) {
         // Navigate to confirmation page on success
         router.push("/sessions/book/confirmation");
+      } else if (response.status === 400) {
+        const responseData = await response.json();
+        if (responseData.message === "Game session at max capacity") {
+          const updatedCart = cart.filter((session) => !session.isFull);
```

Comment:
This will not work. Go the the /sessions page and look at how this data is generated. Invalidating the query doesn't update this data.

Reply:
Instead you might need to compare this data (that's in the cart) against the source data once you update it and remove it based on the differences.